src/commands/registry.rs

```diff
@@ -84,9 +89,11 @@ impl Push {
         notify_if_nondefault_rel(&app_file, distance);
 
         if self.build {
-            spin_build::build_default(&app_file, self.cache_dir.clone()).await?;
+            spin_build::build_default(&app_file, self.profile(), self.cache_dir.clone()).await?;
         }
 
+        spin_build::warn_if_not_latest_build(&app_file, self.profile());
```

Comment:
This seems slightly easier to reason about than relying on the implicit write and read. :shrug: 

```suggestion
        } else {
            spin_build::warn_if_not_latest_build(&app_file, self.profile());
        }
```